java/arcs/core/allocator/Allocator.kt

```diff
@@ -87,12 +118,21 @@ class Allocator(val hostRegistry: HostRegistry) {
             it.hostId == arcHost
         }.firstOrNull() ?: throw ArcHostNotFoundException(arcHost)
 
-    /**
-     * Persists [ArcId] and associated [PlanPartition]s.
-     */
-    private fun writePartitionMap(arcId: ArcId, partitions: List<Plan.Partition>) {
+    /** Persists [ArcId] and associated [PlanPartition]s */
+    private suspend fun writePartitionMap(arcId: ArcId, partitions: List<Plan.Partition>) {
         partitionMap[arcId] = partitions
-        // TODO(cromwellian): implement actual persistence that survives reboot?
+
+        partitions.forEach {
```

Comment:
Optional: for big lambdas like this, I'd suggest giving it a proper variable name. This will let you use `it` in smaller lambdas inside it, where the name is less important.

```
partitions.forEach { partition ->
    ...
    collections["particles"] = partition.particles.map {
        it.particleName.toReferencable()
    }.toSet()
    ...
}
```